src/passes/TypeRefining.cpp

```diff
@@ -400,11 +400,16 @@ struct TypeRefining : public Pass {
 
       void visitStructSet(StructSet* curr) {
         if (curr->type == Type::unreachable) {
+          // Ignore unreachable code.
+          return;
+        }
+        auto type = curr->ref->type.getHeapType();
+        if (!type.isStruct()) {
```

Comment:
This could be slightly more explicit:
```suggestion
        if (type.isBottom()) {
```